site/cds_rdm/inspire_harvester/writer.py

```diff
@@ -0,0 +1,228 @@
+        ]
+
+        differences = len(only_in_entry) + len(only_in_existing)
+        if differences:
+            new_version_success = self._create_new_version(
+                only_in_entry, only_in_existing, inspire_id, entry, existing_rec_id
+            )
+            if not new_version_success:
+                return False
+        else:
+            current_rdm_records_service.edit(system_identity, existing_rec_id)
+            current_rdm_records_service.update_draft(
+                system_identity, existing_rec_id, entry
+            )
+            current_rdm_records_service.publish(system_identity, existing_rec_id)
+
+    def _create_new_version(
+        self, only_in_entry, only_in_existing, inspire_id, entry, existing_rec_id
+    ):
+        """For records with updated files coming from INSPIRE, create and publish a new version."""
+        new_version_draft = current_rdm_records_service.new_version(
+            system_identity, existing_rec_id
+        )
+        draft_id = new_version_draft.to_dict()["id"]
+        files_enabled = new_version_draft.to_dict()["files"]["enabled"]
+
+        if files_enabled:
+            current_rdm_records_service.import_files(system_identity, draft_id)
+
+        if only_in_entry:
+            try:
+                if not files_enabled:
+                    draft = new_version_draft._record
+                    draft.files.enabled = True
+                    draft.commit()
+
+                self._create_files(only_in_entry, draft_id, inspire_id)
+            except WriterError:
+                current_rdm_records_service.delete_draft(system_identity, draft_id)
+                return False
+
+        if only_in_existing:
+            for key in only_in_existing:
+                current_rdm_records_service.draft_files.delete_file(
+                    system_identity, draft_id, key
+                )
+
+        # update metadata as well
+        current_rdm_records_service.update_draft(system_identity, draft_id, entry)
+        current_rdm_records_service.publish(system_identity, draft_id)
+        return True
+
+    def _create_new_record(self, inspire_id, entry):
+        """For new records coming from INSPIRE, create and publish a draft in CDS."""
+        draft = current_rdm_records_service.create(system_identity, data=entry)
+        file_entries = entry["files"].get("entries", None)
+        if file_entries:
+            try:
+                self._create_files(file_entries, draft["id"], inspire_id)
+            except WriterError:
+                current_rdm_records_service.delete_draft(system_identity, draft["id"])
+                return False
+
+        current_rdm_records_service.publish(system_identity, draft["id"])
+        return True
+
+    def _get_existing_records(self, inspire_id):
+        """Find records that have already been harvested from INSPIRE."""
+        # for now checking only by inspire id
+        return current_rdm_records_service.search(
```

Comment:
how many records can search return? are you sure we are getting all of them by using this query?